### PR TITLE
[14.0] [MIG] Migrated fieldservice module.

### PR DESCRIPTION
**Pull request by Murtaza-OSI**

### CODE REVIEW

File: fieldservice/models/fsm_order.py [hunk 1]
```diff
@@ -377,6 +377,8 @@ def onchange_scheduled_date_end(self):
     @api.onchange("scheduled_duration")
     def onchange_scheduled_duration(self):
         if self.scheduled_duration and self.scheduled_date_start:
+            if self.scheduled_duration < 0:
+                raise ValidationError(_("Scheduled Duration Should be Postive Value"))
```
Comment (hparfr) [inline fieldservice/models/fsm_order.py:381]: or = 0 ?

File: fieldservice/models/fsm_person.py [hunk 2]
```diff
@@ -32,6 +34,10 @@ class FSMPerson(models.Model):
     mobile = fields.Char(string="Mobile")
     territory_ids = fields.Many2many("res.territory", string="Territories")
 
+    message_main_attachment_id = fields.Many2one(string="Main Attachment",
```
Comment (hparfr) [inline fieldservice/models/fsm_person.py]: why ?

File: fieldservice/models/res_config_settings.py [hunk 3]
```diff
@@ -88,12 +94,6 @@ class ResConfigSettings(models.TransientModel):
         readonly=False,
     )
 
-    # Dependencies
-    @api.onchange("group_fsm_equipment")
-    def _onchange_group_fsm_equipment(self):
-        if not self.group_fsm_equipment:
-            self.auto_populate_the_equipments = False
-
```
Comment (hparfr) [inline fieldservice/models/res_config_settings.py]: why only these two ?

**Review by hparfr — CHANGES_REQUESTED** [review]
Thanks for this work.

Please also patch security rules.
```xml
<field name="domain_force">['|',('company_id','=',False),('company_id','child_of',[user.company_id.id])]</field>

<field name="domain_force">[('company_id', 'in', company_ids)]</field>

```

File: fieldservice/models/fsm_order.py (hunk 1 above)
Comment (Murtaza-OSI) [reply fieldservice/models/fsm_order.py:381]: The default value is zero so no need to check = into it.

**Comment by hparfr** [comment]
Hi @Murtaza-OSI  do you plan to work on this pr soon ?

**Comment by hparfr** [comment]
Also please integrate the removal of partner_fax https://github.com/OCA/field-service/pull/693

**Comment by brian10048** [comment]
Thank you for the work @Murtaza-OSI 
Are you able to continue work on this PR?

What is reason to remove the two onchange methods in res_config_settings [here](https://github.com/OCA/field-service/pull/695/files/aa9c50aa1464eed32ab055041303e323be224d18..958bd9cd0e94620295bc9cf6c15e5ac343a2ddad#r559497851) ?

Why is the message_main_attachment_id field added in fsm_person [here](https://github.com/OCA/field-service/pull/695#discussion_r559497026) ?

Also can you remove partner_fax like we did in #693 to keep consistency between v13 and v14?

Thank you!
cc: @hparfr @bodedra

**Comment by brian10048** [comment]
Superseded by #740

**Pull request closed without merging**